nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -99,6 +99,18 @@ impl DataStore {
         Ok((sled, was_modified))
     }
 
+    /// Confirms that a sled exists and is in-service.
+    pub async fn check_sled_in_service(
+        &self,
+        opctx: &OpContext,
+        sled_id: Uuid,
```

Comment:
Would be nice to make this a SledUuid, if we can?

(I know we're gonna still probably need to convert to / from the raw UUID, but for new APIs, it would be nice to stick with the strongly-typed variants)

Reply:
Yep, easy enough: 4c6995150